### PR TITLE
abort_on_panic in tests

### PR DESCRIPTION
**Pull request by pompon0**
Made some of the tests abort on panic. We started using "cargo nextest" instead of "cargo test" in CI, so now every test is executed in a separate process and aborting a test won't kill the test harness. This will potentially reveal some flaky tests, which behave in an unexpected way (and panic) but report success anyway. In a long term it is expected to make debugging tests easier, since they will terminate as soon as first panic occurs. Enabling this behavior is per test, since we have some tests which actually expect a panic.

### CODE REVIEW

**Review by nagisa — COMMENTED** [review]
This seems okay to me, though I wonder if there is a way to set `panic="abort"` for `cargo nextest` specifically via `Cargo.toml` and leave `cargo test` workloads intact?

My concern is largely that most developers will be running plain `cargo test` when testing locally and aborting like this is likely going to be a pretty opaque failure mode (though we do invoke an original panic handler – maybe that’s just fine?)

File: chain/network/src/peer_manager/tests/routing.rs [hunk 1]
```diff
@@ -28,9 +28,19 @@ use std::collections::HashSet;
 use std::net::Ipv4Addr;
 use std::sync::Arc;
 
+fn abort_on_panic() {
+    let orig_hook = std::panic::take_hook();
+    std::panic::set_hook(Box::new(move |panic_info| {
+        orig_hook(panic_info);
+        // TODO(gprusak): print stacktrace?
```
Comment (nagisa) [inline chain/network/src/peer_manager/tests/routing.rs:35]: Doesn’t the original hook do so when `RUST_BACKTRACE=1` is set?

**Review by nagisa — APPROVED** [review]

File: chain/network/src/peer_manager/tests/routing.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/peer_manager/tests/routing.rs:35]: yes, it does. thx

**Comment by pompon0** [comment]
@nagisa that's a very good point. In fact if
* tests are executed in the same process (like in "cargo test")
* first one calls abort_on_panic
* second one actually expects a panic (via assert_panic for example)
Then the test suite will fail.

I don't think that detecting in compilation time that a test binary is built under nextest is feasible, because accoding to the documentation nextest doesn't assume that the test binary has been built via "cargo nextest". However it is possible to detect that a process is being run under "cargo nextest" in runtime (via env variables), so I'll use that.

**Pull request merged**